java/c/src/test/java/org/apache/arrow/c/RoundtripTest.java

```diff
@@ -462,6 +462,13 @@ public void testUInt8Vector() {
     }
   }
 
+  @Test
+  public void testNullVector() {
+    try (final NullVector vector = new NullVector("v", 1024)) {
+      assertTrue(roundtrip(vector, NullVector.class));
```

Comment:
Hmm, its not obvious to me why this test is failing. I would've expected your change to `loadFieldBuffers()` to work. Maybe the field comparison is failing? I think you could try passing a custom `TypeEqualsVisitor` with `checkName` or `checkMetadata` disabled to narrow things down on line 155.

```
return VectorEqualsVisitor.vectorEquals(vector, imported, <custom_visitor_here>);
```